### ISSUE TITLE
Rosco: Add subnet/vpc Id options to AWS image virtualizationSettings

### ISSUE THREAD

**Issue by KierranM** [spinnaker/spinnaker#6182]
### Issue Summary: 

To better facilitate multi-region AMI baking in AWS I'd like to be able to specify the VPC/Subnet for a specific region in the virtualizationSettings in my Halyard configuration.

### Description:

Currently we are using one of the workarounds documented in #825, however it would be preferable to be able to specify these settings in our halyard configuration, rather than needing to shim Packer or having a template file per-region.

Ideally I'd like to see the subnet/vpc settings being configurable as options in the `virtualizationSettings` for a base image i.e

```yaml
- baseImage:
     id: ubuntu
     shortDescription: v18.04
     detailedDescription: Ubuntu
     packageType: deb
  virtualizationSettings:
      - region: us-east-1
        virtualizationType: hvm
        instanceType: t3.micro
        sourceAmi: ami-1241234
        sshUserName: ubuntu
        vpcId: vpc-124e124124
        subnetId: subnet-12412414
      - region: us-west-2
        virtualizationType: hvm
        instanceType: t3.micro
        sourceAmi: ami-59396769
        sshUserName: ubuntu
        vpcId: vpc-231241sda
        subnetId: subnet-asd14124qsd12
```

### Steps to Reproduce:

1. Configure AWS provider with a base image with multiple regions that have non-default VPCs
2. Attempt to bake the AMI in parallel in each region
3. Spinnaker will attempt to use non-existent VPCs/Subnets (or the default VPC which may not be appropriate)

### Additional Details:

Packer actually supports specifying subnet_filters/vpc_filters since 1.3.2 (hashicorp/packer#6339), but I believe it would be a better experience if Spinnaker supported setting the subnets/vpc explicitly so that the included packer templates could be made to work multi-region.

**Comment by KierranM** [comment]
I haven't contributed to Spinnaker before, but based on my limited investigations into Rosco's source while writing this I'd be quite keen to try contributing this feature as it doesn't _appear_ to be an overly complicated feature to add.

**Comment by robzienert** [comment]
I'm not part of the Ops SIG, but this seems like a reasonable thing to me - open a PR!

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by kskewes** [comment]
@spinnakerbot remove-label stale

**Comment by spinnakerbot** [comment]
This issue is tagged as 'stale' and hasn't been updated  in 45 days, so we are tagging it as 'to-be-closed'. It will be closed in 45 days unless updates are made. If you want to remove this label, comment:

> @spinnakerbot remove-label to-be-closed

**Comment by spinnakerbot** [comment]
This issue is tagged as 'to-be-closed' and hasn't been updated  in 45 days, so we are closing it. You can always reopen this issue if needed.
